lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2279,15 +2279,17 @@ public function getDefaultValueDeclarationSQL($field)
         if (isset($field['default'])) {
             $default = " DEFAULT '".$field['default']."'";
             if (isset($field['type'])) {
-                if (in_array((string) $field['type'], ["Integer", "BigInt", "SmallInt"])) {
+                $type = $field['type'];
+                if ($type instanceof Types\PhpIntegerMappingType) {
                     $default = " DEFAULT ".$field['default'];
-                } elseif (in_array((string) $field['type'], ['DateTime', 'DateTimeTz']) && $field['default'] == $this->getCurrentTimestampSQL()) {
+                } elseif ($type instanceof Types\PhpDateTimeMappingType
+                    && $field['default'] == $this->getCurrentTimestampSQL()) {
                     $default = " DEFAULT ".$this->getCurrentTimestampSQL();
-                } elseif ((string) $field['type'] == 'Time' && $field['default'] == $this->getCurrentTimeSQL()) {
+                } elseif ($type instanceof Types\TimeType && $field['default'] == $this->getCurrentTimeSQL()) {
                     $default = " DEFAULT ".$this->getCurrentTimeSQL();
-                } elseif ((string) $field['type'] == 'Date' && $field['default'] == $this->getCurrentDateSQL()) {
-                    $default = " DEFAULT ".$this->getCurrentDateSQL();
-                } elseif ((string) $field['type'] == 'Boolean') {
+                } elseif ($type instanceof Types\DateType && $field['default'] == $this->getCurrentDateSQL()) {
```

Comment:
Maybe I should name the marker interface differently because of `Types\DateType` does not implement it although it does map to a PHP `\DateTimeInterface` instance? Suggestions?